cpp/src/replace/replace.cu

```diff
@@ -13,26 +16,551 @@
+                                                                           cudaStream_t stream,
+                                                                           rmm::mr::device_memory_resource* mr) {
+  CUDF_FAIL("Strings not supported for replacement");
+}
+
+
+} //end anonymous namespace
+
+
+namespace cudf {
+namespace detail {
+
+std::unique_ptr<cudf::column> replace_nulls(cudf::column_view const& input,
+                                            cudf::column_view const& replacement,
+                                            rmm::mr::device_memory_resource* mr,
+                                            cudaStream_t stream)
+{
+  if (input.size() == 0) {
+    return std::make_unique<cudf::column>(input);
+  }
+
+  CUDF_EXPECTS(nullptr != input.data<int32_t>(), "Null input data");
+
+  if (input.nullable() == false || input.null_count() == 0) {
+    return std::make_unique<cudf::column>(input);
+  }
+
+  CUDF_EXPECTS(input.type() == replacement.type(), "Data type mismatch");
+  CUDF_EXPECTS(replacement.size() == 1 || replacement.size() == input.size(), "Column size mismatch");
+  CUDF_EXPECTS(nullptr != replacement.data<int32_t>(), "Null replacement data");
+  CUDF_EXPECTS(replacement.nullable() == false || 0 == replacement.null_count(),
+               "Invalid replacement data");
+
+  std::unique_ptr<cudf::column> output = make_numeric_column(input.type(),
+                                                             input.size(),
+                                                             UNALLOCATED,
+                                                             stream,
+                                                             mr);
+  cudf::mutable_column_view outputView = (*output).mutable_view();
+  cudf::experimental::type_dispatcher(input.type(),
+                                      replace_nulls_column_kernel_forwarder{},
+                                      input,
+                                      replacement,
+                                      outputView,
+                                      stream);
+  return output;
+}
+
+
+std::unique_ptr<cudf::column> replace_nulls(cudf::column_view const& input,
+                                            cudf::scalar const* replacement,
+                                            rmm::mr::device_memory_resource* mr,
+                                            cudaStream_t stream)
+{
+  if (input.size() == 0) {
+    return std::unique_ptr<cudf::column>(new cudf::column(input));
+  }
+
+  CUDF_EXPECTS(nullptr != input.data<int32_t>(), "Null input data");
+
+  if (input.null_count() == 0 || input.nullable() == false) {
+    return std::unique_ptr<cudf::column>(new cudf::column(input));
+  }
+
+  CUDF_EXPECTS(input.type() == replacement->type(), "Data type mismatch");
+  CUDF_EXPECTS(true == replacement->is_valid(stream), "Invalid replacement data");
+
+  std::unique_ptr<cudf::column> output = make_numeric_column(input.type(),
```

Comment:
This is not correct. What about timestamp columns?

I suggest just making a copy of `input` and then replacing the null values.

Reply:
Done.